### PR TITLE
Feature/transfer process initialization

### PR DESCRIPTION
**Pull request by davidjovanovic**

### CODE REVIEW

**Review by IgorBalog-Eng — COMMENTED** [review]
Review

File: catalog/src/test/java/it/eng/catalog/util/MockObjectUtil.java [hunk 1]
```diff
@@ -48,6 +48,7 @@ public class MockObjectUtil {
     public static final Instant MODIFIED = Instant.parse("2024-04-23T16:26:00Z");
     public static final String TITLE = "Title for test";
     public static final String ENDPOINT_URL = "https://provider-a.com/connector";
+    private static final String FILE_ID = "some file";
```
[the remark this answers is not in the document]
Comment (davidjovanovic) [reply catalog/src/test/java/it/eng/catalog/util/MockObjectUtil.java:51]: Should I remove the spaces in all constants in that class?

File: catalog/src/main/java/it/eng/catalog/service/DatasetService.java [hunk 2]
```diff
@@ -121,4 +124,20 @@ public Dataset updateDataset(String id, Dataset dataset) {
 
         return storedDataset;
     }
+
+	public List<String> getFormatsFromDataset(String id) {
```
[the remark this answers is not in the document]
Comment (davidjovanovic) [reply catalog/src/main/java/it/eng/catalog/service/DatasetService.java:128]: Good that you noticed, I added for the controller but not for the service.

File: catalog/src/main/java/it/eng/catalog/rest/api/DatasetAPIController.java [hunk 3]
```diff
@@ -42,6 +43,24 @@ public ResponseEntity<GenericApiResponse<JsonNode>> getDatasetById(@PathVariable
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON)
                 .body(GenericApiResponse.success(Serializer.serializePlainJsonNode(dataset), "Fetched dataset"));
     }
+    
+    @GetMapping(path = "/{id}/formats")
```
[the remark this answers is not in the document]
Comment (davidjovanovic) [reply catalog/src/main/java/it/eng/catalog/rest/api/DatasetAPIController.java:47]: Will do 👍

File: data-transfer/src/main/java/it/eng/datatransfer/service/api/RestArtifactService.java [hunk 4]
```diff
@@ -53,7 +65,10 @@ public String getArtifact(String transactionId, JsonNode jsonBody) {
 	
 	public GridFsResource streamAttachment(String transactionId) {
 		TransferProcess transferProcess = getTransferProcessForTransactionId(transactionId);
-		String fileId = transferProcess.getFileId();
+		GenericApiResponse<String> response = okHttpRestClient.sendGETRequest("http://localhost:" + properties.serverPort() + ApiEndpoints.CATALOG_DATASETS_V1 + "/fileid",
```
[the remark this answers is not in the document]
Comment (davidjovanovic) [reply data-transfer/src/main/java/it/eng/datatransfer/service/api/RestArtifactService.java:68]: Thanks 👍

File: catalog/src/test/java/it/eng/catalog/util/MockObjectUtil.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (IgorBalog-Eng) [reply catalog/src/test/java/it/eng/catalog/util/MockObjectUtil.java:51]: To be 110% compatible, you can check 
https://eclipse-dataspace-protocol-base.github.io/DataspaceProtocol/2024-1/common/schema/context.json
and see which fields are of type @id, and change only those

File: catalog/src/test/java/it/eng/catalog/service/DatasetServiceTest.java [hunk 5]
```diff
@@ -42,23 +44,16 @@ public class DatasetServiceTest {
...
-    }
+    private Dataset dataset = MockObjectUtil.DATASET_WITH_FILE_ID;
     
-    @Test
-    @DisplayName("Get dataset by ID throws exception when not found")
-    void getDataSetById_notFound() {
-        when(repository.findById(anyString())).thenReturn(Optional.empty());
-        assertThrows(CatalogErrorException.class, () -> datasetService.getDatasetById("datasetId"));
-    }
+    private Dataset datasetWithoutDistributions = Dataset.Builder.newInstance()
```
Comment (IgorBalog-Eng) [inline catalog/src/test/java/it/eng/catalog/service/DatasetServiceTest.java:49]: Nice one to include those usecases

**Review by IgorBalog-Eng — APPROVED** [review]

**Review by marest94 — APPROVED** [review]

**Pull request merged**